dhall/src/Dhall/TypeCheck.hs

```diff
@@ -2769,12 +2786,58 @@ prettyTypeMessage (InvalidRecordCompletion fieldName expr0) = ErrorMessages {..}
+        \                                                                                \n\
+        \                                                                                \n\
+        \    ┌────────────────────────────────┐                                          \n\
+        \    │ { x = { y = 1 } } with x.y = 2 │                                          \n\
+        \    └────────────────────────────────┘                                          \n\
+        \                                                                                \n\
+        \                                                                                \n\
+        \    ┌────────────────────────────────────────────────────────────┐              \n\
+        \    │ λ(r : { foo : { bar : Bool } }) → r with foo.bar = False } │              \n\
+        \    └────────────────────────────────────────────────────────────┘              \n\
+        \                                                                                \n\
+        \                                                                                \n\
+        \... but you cannot update values that are not records.                          \n\
+        \                                                                                \n\
+        \For example, the following expression is " <> _NOT <> " valid:                  \n\
+        \                                                                                \n\
+        \                                                                                \n\
+        \    ┌─────────────────┐                                                         \n\
+        \    │ 1 with x = True │                                                         \n\
+        \    └─────────────────┘                                                         \n\
+        \      ⇧                                                                         \n\
+        \      Invalid: Not a record                                                     \n\
+        \                                                                                \n\
+        \                                                                                \n\
+        \────────────────────────────────────────────────────────────────────────────────\n\
+        \                                                                                \n\
+        \You tried to update this expression:                                            \n\
+        \                                                                                \n\
+        \" <> insert expression <> "\n\
+        \                                                                                \n\
+        \... which is not a record, but is actually a:                                   \n\
+        \                                                                                \n\
+        \" <> insert typeExpression <> "\n"
```

Comment:
Since the problematic `with`-expression may itself be the result of desugaring, I think it might be useful to include the full offending expression.

E.g. in the case of `{ a = 1 } with a.b = 2` we currently produce

```
You tried to update this expression:

↳ 1

... which is not a record, but is actually a:

↳ Natural
```

If we include the full `1 with b = 2` that might help explain how the desugaring works and how the problem arises.